apps/twitch-bot/internal/command/addcommandalias.go

```diff
@@ -0,0 +1,63 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/gempir/go-twitch-irc/v3"
+	"github.com/senchabot-dev/monorepo/apps/twitch-bot/client"
+	"github.com/senchabot-dev/monorepo/apps/twitch-bot/server"
+)
+
+const ADD_COMMAND_ALIAS_INFO = "!acmda [command_name] [command_alias(es) separated by space]"
+
+func AddCommandAliasCommand(client *client.Clients, server *server.SenchabotAPIServer, message twitch.PrivateMessage, commandName string, params []string) {
+	if !strings.Contains(message.Tags["badges"], "moderator") {
+		if !strings.Contains(message.Tags["badges"], "broadcaster") {
```

Comment:
Let's move this to a helper to make it more readable and also reusable.

```go
func canExecuteCommand(message twitch.PrivateMessage) bool {
  // let's actually convert the logic to more understandable one
  
  // moderator can run the command
  if strings.Contains(message.Tags["badges"], "moderator") {
    return true
  }
  
  // broadcaster can run the command
  if strings.Contains(message.Tags["badges"], "broadcaster") {
    return true
  }
  
  // everyone else cannot run the command.
  return false
}
```

WDYT?

Reply:
I'd probably go even one step further and implement `func isBroadcaster(message twitch.PrivateMessage)` and `func isModerator(...)` to make things even more reusable.